### DISCUSSION TITLE
Campaign/design feedback: Feature creep, mechanic dissonance, gameplay unity.

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#12707] [Unstable feedback]
### The expansion of the world in gameplay and it's mechanics is good and I like the ideas, though in execution and balance I found the major additions in recent updates to be rather lukewarm reception among myself and others.

To give comparison: In the early versions of the game there was only character-skills, crafting/medicine-crafting, and medical-treatment.  Things were very simple - yes - but very mechanically tight, and these three things were not only connected but interacted with each-other smoothly as a larger gameplay system.

Now compare this to factions, genetics, perks/talents, submarine upgrade tiers. You end up having to juggle >5 different mechanical systems, and they don't feel like they work together with each-other. It can end up feeling bloated because of this, and you can often completely forget about a mechanic such as genetics because it only exists and operates within it's own tiny niche. These mechanics generally feel like disparate self-contained parts that exist and serve their own function well enough but don't really contribute their parts to the the sum in totality. To give example of what I mean:

- Talents feel like an unrelated part of character stats & progression from their base skills & equipment. It also kind of makes skills irrelevant as now 'EXP' is all that seems to matter (and lots of talents just let you *"BUY"* skill points with them). It seems to conflict or fly in the face of skill-levels rather than compliment it in some way. Just thinking out loud: but for instance instead of having EXP independent of stats maybe instead class traits unlock as your skills improve - or the reverse and have your character stats directly 'level up' when your EXP does and you pick your trait and fold the old skill system stats into this new progression.

- Talents could be more focused on personal character ability, less 'meta' effect talents (exceptions being when it makes sense in it's context like captain's command talents), or perhaps have submarine-wide 'talents' be a part of some kind of global progression which are talents attached to your campaign as a whole (instead of a single person), IE the 'talents' of the vessel (you could frame this any number of ways, like a submarine's 'reputation' / 'commendations' as you go from being a rag-tag crew of beginners to a submarine & crew with exceptional history & story. Plus this would sort of allow a feeling of permanent progression over the course of the campaign even as singleplayer characters are lost or players join & leave.)

- Recipes gained by talents is a weird issue, alot of the talent items are so "OP"/"quality-of-life"/"essential" that most players will always just pick the talent tree which gives them their class item recipes hands down. The progression of obtaining these class items or unique faction items could have been done more organically implemented into gameplay progression: buying blueprints from a station, buying blueprints from a faction, looting blueprints from wrecks or ruins, a 'Research' system, progression-based fabricator 'upgrades'??? the list goes on. Plus from a meta perspective the fact that your character just 'suddenly knows' how to do a recipe always seemed a bit contrived and under-thought. 

Talents as a whole feel sort of "gamey" because of this disconnect - they don't really feel like a part of the world, they just feel like buttons you press in GUI to unlock some recipe or status effect when it could have been something more involved in the experience of the characters and the interaction with the world.

- Genetics really exists in it's own bubble, for one it's reliant on the talent's of the medic (which I can sort of understand why it's not very deep in complexity since it was just made for this one thing). As a mechanic its not really impacted much/at-all by the other aspects of the medic 'gameplay loop' - skills or medical fabrication doesn't really matter for genetics, only the talent tree that directly improves refined genetic material & improves splicer. The decision to require players use research tables on stations I don't really have any direct criticisms with (it can be annoying trying to find a station with a research table, or late-campaign when there literally isn't any outpost anymore; but both those issues can be solved if players choose to with custom submarines.) Although it does feel lacking that there's not really anything for medics to do otherwise in terms of maybe being able to work with some part of the genetics system while onboard the ship doing their other duties, not sure what that would entail specifically though (More in-depth genetic research system?). Additionally as most players have pointed out only really 4 of the genes are super useful, also the potential downsides are so debilitating to deal with and the fact that genetic material can't be extracted after being spliced means there's no point in splicing genes until they are at 100% purity and no side-effects. 

Either way, genes really play so little part in the grand scheme of the campaign because you spend most of the time just collecting samples and waiting until you can research them, and then waiting until you've refined the genes to 100%, and waiting until your medic unlocks the better splice - then you just set your gene splicer and forget about it since there's nothing else to do with genetics once you pick what 2 of the 4 OP abilities you want and install them. Would have been nice to see genetics play more into the benefits and drawbacks of the other medical systems as opposed to being this weird little extra thing that was just thrown in.

- Submarine tiers, classes, and upgrade progression: This one doesn't actually pertain to the other mechanics much but I just wanted to criticize it anyways... I'd prefer a more 'open ended' upgrade progression like what we initially had over this class & tier enforced system - on that note I thought the elimination of 'deep divers' was a good step in that direction by allowing **ANY SUBMARINE** to be taken to the end of the game if sufficiently upgraded - until we got restricted submarine upgrades and restricted submarine tiers and we're back to square one - like I said: *"Tier-3 is just deep diver again"*. Like if somebody has the sufficient materials (I'm sort of mixed on the fact that upgrades take materials now, because of underlying material economy balance issues - as well as the fact it kind of doesn't make sense in-universe that a shipyard you are paying for wouldn't be including the parts/resources of their own in the cost you're paying for, for certain things like rare alien materials it makes sense but maybe that should only be for very late-game upgrades) and credits they should be within' their rights to turn even an R-29 into a late-game speedy warship if they *really want to*.
Going to my previous points, the mechanic perk that lets you take these additional upgrade levels could instead be included in your persistent "submarine's talents/perks/reputation" - again so it's not 'tied to a single character'. Plus as has already been brought up, the selection of submarines in vanilla is already very limited, so fracturing that selection further across multiple specific classes and tiers means there's very little freedom of choice within' a specific class a player wants during each stage of campaign. 

Additionally the issue of crew sizes, since a small crew from a dugong isn't going to be able to cope easily with being forced into operating a Kastrull or a Remora, conversely if you've just started a campaign and you have a full 16 player crew you're not going to be happy crammed like sardines into a Barsuk. So if submarine tiers and classes are fairly permanent the way they are, you could really use alot more variety of vanilla submarines with different sizes and classes to properly fill the full gamut of submarine-tiers for each gameplay style the player could possibly choose, IE: "I want a large T-1 transport ship that's underequipped but plenty of room for the crew" or "I like my small scout but I'm at the late game so I need a well-equipped T-3 that isn't too large"

**Comment by Hazmat616** [comment]
What if you could progress all trees simultaneously but each tree has a specific way to earn xp for it? For example the only way to earn xp for frog man is to earn xp or skills in water. Or using a turret for gunner.

**Comment by bookslayer10** [comment]
I agree with everything you've said. While I'm still happy playing with the systems you have mentioned, improvements would be welcome.

What if experience and skill gain were tied into the same system? Instead of repairing devices or shooting monsters giving you skill and beating missions giving you experience, they could be rolled into one. Every time you would get a skill point, you instead gain some experience.

Depending on the class, certain actions give more or less XP; gunners gain more experience for shooting monsters, electricians gain more experience for fixing junction boxes. Or maybe you gain XP based on how much skill you have with the task you are doing.

Level up, and you can spend XP to unlock more skill points, or on talents as usual. As you buy more skill points, it costs more XP for each point, so reaching level 100 would be hard to do for multiple skills.

**Comment by jigardi** [comment]
Agree with OP on most of the points raised here.  And would add that XP gain being tied only to mission success is in itself a colossal mistake,  massively compounding the points OP raised. Case in point:

I've never seen anyone want to continue when a mission fails.  It's always automatically assumed to be grounds for an insta-reload.

In my experience,  it generally requires herculean efforts to convince players to keep going in such situations.

The attempt is virtually guaranteed to be an exercise in futility.  Most players will instantly deflate on Mission Failure, leading to curt or even flat out angry refusals, when asked to spare even a second's thought to anything other than the expected insta-reload.

If by some miracle you do manage to convince them,  almost all potential rewards for their additional efforts have already been forfeited by the MissionFailure, so they'll only ever press on begrudgingly.  Any skills gained by the remaining survivors are most likely outweighed by the skill loss of the dead.  Overall it's just a bad time for everyone involved.

I can't even fault them for reacting/thinking in that manner:  _this pervasive defeatist attitude is the naturally correct response/stance to the current Mission/Payment/XP system._

When you factor in what OP mentioned about certain recipe-talents being too good to pass up, it also means people don't even get to pick the talents they actually want to go for.  Instead they have to slave away for fickle XP which they will then have to use largely on things they don't actually want.  It's not a good system and it definitely warps the playerbase.

I know skill loss has settings now,  _but there's nothing you can do about NOT getting XP for a failed mission._

It's a really odd situation in terms of gameplay and gamefeel too.  It's not technically a game-over state like when all the players die at once,  but in praxis it is since the default reaction/stance is to just insta-reload anyway.

...

Depending on how badly a mission fails,  getting less pay or even no pay makes perfect sense, since it tracks roughly with the IRL human experience  -  i.e. there is some positive relation between the perceived value of your work,  and the pay you receive for doing it.

However,  not getting any experience for surviving and persisting - especially when everything has gone maximally wrong - is not just extremely unsatisfying,  it makes absolutely no sense and does not track with the IRL human experience at all - i.e. experience is not transferred to us instantaneously when we receive our paycheck!  It's very much its own and altogether separate thing!

Experience would be much better described as a trickling U.B.I./stipend,  which you are always receiving so long as you are conscious~/~alive!

I think that, THAT would make for a much better _experience_ experience, than what we have now.

It wouldn't even have to break any XP balancing,  since you could just make it cap at whatever the XP for the missions is currently.

...

As is,  the game is basically saying something to the effect of this to the player when a mission fails;

_If you actually like this gameplay, this experience, for a second longer than I say you should;  then I the game will spite you for it._

_Problem solving, effort, and even experience itself, have nothing to do with XP;  only I the game decide what XP is._

_When I the game say your mission has failed,  you should instantly give up and restart;  refuse and I will decrease your crews' interest in the game in direct proportion to your defiance of my command,_

_Untill you learn to do this perfectly,  I will continue to waste your time._

_And when you finally do learn to do this perfectly,  I'll sometimes release this RNG Sword of Damocles looming over your head, and waste your time anyway!_

^If that is the actual design intent, i.e. unapologetic hostility towards the player and disrespect of their invested time and effort, then there's obviously nothing to be done about it,  and we should all just shrug and move on.

However,  if that is not the actual design intent,  but rather a legacy darling that should have been thrown overboard years ago,  then I think there is much to be gained by changing the Mission/Payment/XP-system post-haste.

The arbitrary linkage/analogy between Mission/Payment/XP wouldn't be so absurd and opaque anymore - and the ever-looming threat of the RNG Sword of Damocles wouldn't guarantee an insta-reload as often.

If the design intent of the XP nullification on MissionFailure is to give players a heightened sense of investment in each mission/play-session,  then I think the design has totally backfired;  because while it might initially succeed at heightening the tension/sense of investment, that effect is completely overshadowed in the long run by how it makes people lose interest in the game.

After a certain threshold the excitement simply becomes frustration, then disappointment and eventually absolute resignation.

A more forgiving XP system can't fix that broken design, but it would make for a much more enjoyable experience much more of the time.

What the player chooses to do from moment to moment would continue to matter to the player - even if it was but an illusion of control - the players would continue to feel the weight of their agency up to the last second that they either succeeded or died.

As it stands now,  I feel like most crews just run on the fumes of rapidly depleting tanks of hopium, copium and rage! xD
